src/Messages.sol

```diff
@@ -156,24 +156,25 @@ library ConnectorMessages {
      * 
      * 0: call type (uint8 = 1 byte)
      * 1-8: poolId (uint64 = 8 bytes)
-     * 9-25: trancheId (16 bytes)
-     * 26-46: user (Ethereum address, 20 bytes)
-     * 47-78: amount (uint256 = 32 bytes)
-     * 
+     * 9-24: trancheId (16 bytes)
+     * 25-56: user (Ethereum address, 20 bytes - Skip last 12 bytes for 32-byte address compatibility)
+     * 57-72: amount (uint128 = 16 bytes)
+     * 73-82: domain (Domain = 9 bytes)
      */
-    function formatTransfer(uint64 poolId, bytes16 trancheId, address user, uint256 amount, bytes9 destinationDomain) internal pure returns (bytes memory) {
-        return abi.encodePacked(uint8(Call.Transfer), poolId, trancheId, user, amount, destinationDomain);
+    function formatTransfer(uint64 poolId, bytes16 trancheId, address user, uint128 amount, bytes9 destinationDomain) internal pure returns (bytes memory) {
```

Comment:
We needed to encode amount as `uint128`

Reply:
I'm a little unsure of using a `u128` for `amount` since balances are generally `u256` in solidity, but I don't know if it really matters. If we run into issues later we can always change it.

Reply:
I remember talking about this with @offerijns and the conclusion was that u128 was enough for all operations we need. u128 is the type used in the cent-chain so that type should be big enough to cover tranche transfers in the EVM as well 👍